openpathsampling/dynamics_engine.py

```diff
@@ -296,6 +294,9 @@ def generate(self, snapshot, running=None):
             logger.info("Starting trajectory")
             log_freq = 10 # TODO: set this from a singleton class 
             while stop == False:
+                if self.options.get('n_frames_max', None) is not None :
```

Comment:
Yes, this is a fail safe. Right now all dynamic engines must contain this option and if we keep it that way we can remove since in the constructor it will be checked.